### PR TITLE
Non code tweaks

### PR DESCRIPTION
**Pull request by elliott-imhoff**
# Some small updates to non-code files
## What?
* Added ``configuration`` module to auto-generated documentation
* Updated README with documentation generation instructions
* Travis installs via ``pipenv install --dev --deploy``
* Updated PR template
## Why?
These kinds of things really bug me if they aren't taken care of.
## PR Checklist
- [x] Merged latest master
- [x] Updated version number
## Breaking Changes
Nosiribubski

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ python: 3.7
 
 install:
   - pip install pipenv
-  - pipenv install --dev
+  - pipenv install --dev --deploy
```
Comment (elliott-imhoff) [inline .travis.yml]: I prefer have the ``--deploy`` flag on Travis installs to make Travis fail if ever the Pipfile and Pipfile.lock are out of sync.

File: pull_request_template.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # <Title_Here>
 ## What?
 ## Why?
+## PR Checklist
```
Comment (elliott-imhoff) [inline pull_request_template.md]: People have been putting a heading here, which I like.  Might as well automatically put it there.  Let me know if you prefer the "QA Strategy" heading instead.

File: README.md [hunk 3]
```diff
@@ -21,11 +21,10 @@ Else to get started with examples right away, simply follow the *Examples* instr
 
 ### Documentation
 
-This library is documented using Sphinx. To generate documentation, within the top level of
-the repo, run
+This library is documented using Sphinx. To generate documentation, make sure all **dev** 
+dependencies are installed (e.g. you've installed via ``pip install -e .[dev]``).  Within
+the *docs/* subdirectory, run:
```
Comment (elliott-imhoff) [inline README.md]: See #14.  We should use Github Pages to automatically generate and publish documentation.  I've always wanted to do it, but it's only allowed for public repos.

Comment (jdalt) [reply README.md]: How do we get Github Pages automatically running?

Comment (elliott-imhoff) [reply README.md]: It shouldn't be too hard.  There are tons of tutorials online

**Pull request merged**